test/parallel/test-os.js

```diff
@@ -204,11 +220,20 @@ assert.ok(pwd.homedir.includes(path.sep));
 assert.strictEqual(pwd.username, pwdBuf.username.toString('utf8'));
 assert.strictEqual(pwd.homedir, pwdBuf.homedir.toString('utf8'));
 
-// Test that the Symbol.toPrimitive functions work correctly
-[
-  [`${os.hostname}`, os.hostname()],
-  [`${os.homedir}`, os.homedir()],
-  [`${os.release}`, os.release()],
-  [`${os.type}`, os.type()],
-  [`${os.endianness}`, os.endianness()]
-].forEach((set) => assert.strictEqual(set[0], set[1]));
+assert.strictEqual(String(os.hostname), os.hostname());
+assert.strictEqual(String(os.homedir), os.homedir());
+assert.strictEqual(String(os.release), os.release());
+assert.strictEqual(String(os.type), os.type());
+assert.strictEqual(String(os.endianness), os.endianness());
+assert.strictEqual(String(os.tmpdir), os.tmpdir());
+assert.strictEqual(String(os.arch), os.arch());
+assert.strictEqual(String(os.platform), os.platform());
+
+assert.strictEqual(Number(os.totalmem), os.totalmem());
```

Comment:
Nit: I think it is easier to read to stick to the template strings and to `+` for strings. But this is not blocking.